manager/manager.go

```diff
@@ -15,131 +15,46 @@
 package manager
 
 import (
-	"fmt"
-	"sync"
 	"time"
 
-	"github.com/golang/glog"
-
-	"k8s.io/heapster/model"
+	"k8s.io/heapster/aggregator"
 	"k8s.io/heapster/sinks"
-	sink_api "k8s.io/heapster/sinks/api"
-	"k8s.io/heapster/sinks/cache"
-	source_api "k8s.io/heapster/sources/api"
+	"k8s.io/heapster/sources"
 )
 
 // Manager provides an interface to control the core of heapster.
 // Implementations are not required to be thread safe.
 type Manager interface {
-	// Export the latest data point of all metrics.
-	ExportMetrics() ([]*sink_api.Point, error)
-
-	// Set the sinks to use
-	SetSinkUris(Uris) error
-
-	// Get the sinks currently in use
-	SinkUris() Uris
-
-	// Get a reference to the cluster entity of the model, if it exists.
-	GetModel() model.Model
-
-	// Starts the manager.
 	Start()
-
-	// Stops the manager.
 	Stop()
 }
 
 type realManager struct {
-	sources       []source_api.Source
-	cache         cache.Cache
-	model         model.Model
-	modelDuration time.Duration
-	sinkManager   sinks.ExternalSinkManager
-	sinkUris      Uris
+	sources       sources.SourceManager
+	aggregator    aggregator.Aggregator
+	sinks         sinks.SinkManager
 	lastSync      time.Time
 	resolution    time.Duration
-	decoder       sink_api.Decoder
-	mainStopChan  chan struct{}
-	sinkStopChan  chan<- struct{}
-	modelStopChan chan struct{}
-}
-
-type syncData struct {
-	data  source_api.AggregateData
-	mutex sync.Mutex
+	stopChan      chan struct{}
 }
 
-func NewManager(sources []source_api.Source, sinkManager sinks.ExternalSinkManager, res, bufferDuration time.Duration,
-	c cache.Cache, useModel bool, modelRes, modelDuration time.Duration) (Manager, error) {
-
-	var newModel model.Model
-	if useModel {
-		newModel = model.NewModel(modelRes)
-		// Temporary semi-hack to get model storage garbage-collected.
-		c.AddCacheListener(newModel.GetCacheListener())
-	}
-
+func NewManager(sources sources.SourceManager, aggregator aggregator.Aggregator, sinks sinks.SinkManager, res time.Duration) (Manager, error) {
```

Comment:
There should be a list of DataBatchProcessors, that are fired one after another.